std/fiat-shamir/transcript.go

```diff
@@ -110,19 +119,34 @@ func (t *Transcript) ComputeChallenge(challengeID string) (frontend.Variable, er
 	t.h.Reset()
 
 	// write the challenge name, the purpose is to have a domain separator
-	cChallenge := []byte(challengeID) // if we send a string, it is assumed to be a base10 number
-	if challengeName, err := constant.HashedBytes(t.api, cChallenge); err == nil {
-		t.h.Write(challengeName)
+	challengeInput := []byte(challengeID)
+	var challengeHashInput frontend.Variable = challengeInput
+	if t.config.withDomainSeparation {
```

Comment:
So in gnark-crypto we will need to have a fiatShamirConfig also at some point to tune the domain separation variable ? Currently it's hardcoded in WriteString

Reply:
Yes, I do think it would be better to be able to configure it in gnark-crypto also. Right now the domain separation is enforced automatically with a fixed string, but this would prevent (simple) compatiblity with other software. We could have it by wrapping MiMC in gnark-crypto to remove `WriteString` method (replacing it), but imo it is unintuitive. Maybe we can even do an option:
```go
func WithDomainSeparationString(prefix string)
```
or something like that.

Reply:
Created issue: https://github.com/Consensys/gnark-crypto/issues/467